### PR TITLE
Iframes with missing redirect location should fire load event and queue RT entry

### PR DESCRIPTION
**Pull request by noamr**
Iframes with missing redirect location (e.g. 302 response code and no `location` header) should fire load event and queue RT entry

### CODE REVIEW

File: resource-timing/iframe-bad-redirect.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+    promise_test(async t => {
+        const frame = document.createElement('iframe');
+        t.add_cleanup(() => frame.remove());
+        await new Promise(resolve => {
+            frame.addEventListener('load', resolve);
+            frame.src = 'resources/bad-redirect.py';
```
Comment (yoavweiss) [inline resource-timing/iframe-bad-redirect.html:13]: Nit: Maybe rename to redirect-without-location.py or add a comment as to why this redirect is bad?

File: resource-timing/iframe-bad-redirect.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            document.body.appendChild(frame);
+        });
+
+        const entries = performance.getEntriesByName(frame.src);
```
Comment (yoavweiss) [inline resource-timing/iframe-bad-redirect.html:17]: Does this work? I thought `getEntriesByName` needs the absolute URL..

Comment (noamr) [reply resource-timing/iframe-bad-redirect.html:17]: Yes! The `src` attribute returns the full URL

File: resource-timing/iframe-bad-redirect.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Test the sequence of events when reporting iframe timing.</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+    promise_test(async t => {
```
[the remark this answers is not in the document]
Comment (noamr) [reply resource-timing/iframe-bad-redirect.html:8]: The latter is tested elsewhere.

**Review by yoavweiss — APPROVED** [review]
LGTM

**Pull request merged**